include/fmt/format.h

```diff
@@ -3950,6 +3950,26 @@ class formatter<std::basic_string<Char, Traits, Allocator>, Char>
 template <typename Char, size_t N>
 struct formatter<Char[N], Char> : formatter<basic_string_view<Char>, Char> {};
 
+#if FMT_USE_BITINT
+namespace detail {
+template <class T> struct bit_int_format_as {
+  using traits = bitint_traits<T>;
+  static constexpr size_t N = traits::number_of_bits;
+  static constexpr bool is_signed = traits::is_signed;
+  using type = conditional_t<
+      N <= 64, conditional_t<is_signed, int64_t, uint64_t>,
+      conditional_t<(N > 64 && N <= 128),
+                    conditional_t<is_signed, __int128, unsigned __int128>,
+                    void>>;
```

Comment:
I think this can be moved to `bitint_traits` which would make `number_of_bits` and `is_signed` unnecessary.

Reply:
> I think this can be moved to `bitint_traits` which would make `number_of_bits` and `is_signed` unnecessary.

Nice suggestion, have done. Only thing to note, we must `#include <cstdint>` in `base.h` now, for the `int64_t`, if this is acceptable

Reply:
Let's use `(unsigned) long long` which is guaranteed to be at least 64-bit and avoid additional include.

Reply:
> Let's use `(unsigned) long long` which is guaranteed to be at least 64-bit and avoid additional include.

Good shout, have made this change